cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -448,9 +449,12 @@ struct MinMaxImpl : public ScalarAggregator {
     return Status::OK();
   }
 
-  Status ConsumeArray(const ArrayType& arr) {
+  Status ConsumeArray(const ArraySpan& arr_span) {
     StateType local;
 
+    // TODO(wesm): do not use ToArrayData
+    ArrayType arr(arr_span.ToArrayData());
```

Comment:
Is this something that you plan to do in a later PR? Or can it just be done here?